### PR TITLE
DE21691: Navbar is displayed inside ASV for activities

### PR DESCRIPTION
**Pull request by tarynm**
* the iFrame calls source that is returning a navbar since the LMS
expects a navbar and doesn't recognize that the call is from an iframe

### CODE REVIEW

**Comment by tarynm** [comment]
Still a work in progress

**Comment by tarynm** [comment]
@jstefaniuk-d2l I made some changes that seem to stop the navigation from briefly appearing after the Quiz Summary page. Can you take a look whenever you get a chance?

File: src/react-valence-ui-iframe.js [hunk 1]
```diff
@@ -17,6 +17,10 @@ var ResizingIframe = React.createClass({
 	},
 	componentDidMount: function() {
 		this.updateProgress(0);
+		this.updateNavbarStyle();
```
Comment (jstefaniuk-d2l) [inline src/react-valence-ui-iframe.js]: My guess is that this call has no effect

File: src/react-valence-ui-iframe.js [hunk 2]
```diff
@@ -44,6 +48,26 @@ var ResizingIframe = React.createClass({
 			}
 		}
 	},
+	updateNavbarStyle: function() {
+		// Hide the navbar and minibar from within the iframe if it is rendered
+		var iframe = React.findDOMNode(this.refs.iframe);
+		if (iframe) {
```
Comment (jstefaniuk-d2l) [inline src/react-valence-ui-iframe.js]: The iframe will always be in this element, so you don't need to check for it

File: src/react-valence-ui-iframe.js [hunk 3]
```diff
@@ -44,6 +48,26 @@ var ResizingIframe = React.createClass({
...
+			// Can only manipulate the iframe if it is not cross domain
+			if (!ResizeCallbackMaker.crossDomain(iframe)) {
```
Comment (jstefaniuk-d2l) [inline src/react-valence-ui-iframe.js]: If you set a `sameDomain` state value in the handleOnLoad (using the `security` value in the result object), we can avoid calling this a bunch

Comment (jstefaniuk-d2l) [inline src/react-valence-ui-iframe.js]: Actually, if they're changing pages we want to track the new crossDomain value, ignore that comment

File: src/react-valence-ui-iframe.js [hunk 4]
```diff
@@ -17,6 +17,10 @@ var ResizingIframe = React.createClass({
 	},
 	componentDidMount: function() {
 		this.updateProgress(0);
+		this.updateNavbarStyle();
+	},
+	componentDidUpdate: function() {
+		this.updateNavbarStyle();
```
Comment (jstefaniuk-d2l) [inline src/react-valence-ui-iframe.js]: If you can check here (or in the updateNavbarStyle) to see if the iframe changed url before running the update, that would prevent this function from running many times on the same page

**Comment by jstefaniuk-d2l** [comment]
I like this solution, it may be good to check that update isn't getting called too much, but that's my only worry

**Comment by tarynm** [comment]
My last commit description got cut off it seems. I updated the code to include a check on the iframe location before updating.

File: src/react-valence-ui-iframe.js [hunk 5]
```diff
@@ -44,6 +48,28 @@ var ResizingIframe = React.createClass({
...
+					var iframeStyle = document.createElement('style');
+					iframeStyle.innerHTML = '.d2l-navbar, .d2l-minibar-placeholder {display:none;}';
+					iframeStyle.type = 'text/css';
+					var head = iframeDocument.head;
+					if (head) {
+						head.appendChild(iframeStyle);
+					}
+				}
+			}
+			this.state.ifreamLocation = iframe.contentWindow.location.href;
```
Comment (jstefaniuk-d2l) [inline src/react-valence-ui-iframe.js]: use setState here, also spelling

File: src/react-valence-ui-iframe.js [hunk 6]
```diff
@@ -44,6 +48,28 @@ var ResizingIframe = React.createClass({
 			}
 		}
 	},
+	updateNavbarStyle: function() {
+		// Hide the navbar and minibar from within the iframe if it is rendered
+		var iframe = React.findDOMNode(this.refs.iframe);
+		// Only update if the frame location has changed
+		if (this.state.iframeLocation !== iframe.contentWindow.location.href) {
```
Comment (jstefaniuk-d2l) [inline src/react-valence-ui-iframe.js]: Does this throw an error if cross domain?

File: src/react-valence-ui-iframe.js (hunk 5 above)
Comment (tarynm) [inline src/react-valence-ui-iframe.js]: oops, thanks!

File: src/react-valence-ui-iframe.js (hunk 6 above)
Comment (tarynm) [inline src/react-valence-ui-iframe.js]: That's a good point. It should check whether it is cross domain before checking the location here to make sure it has access to the contentWindow.

**Comment by awikkerink** [comment]
It feels a little hacky, but there isn't a much better way of doing it. :+1:

**Comment by awikkerink** [comment]
Tests?

**Comment by tarynm** [comment]
I will add some tests

**Comment by jstefaniuk-d2l** [comment]
Also, I would add a note to the readme mentioning that iframed brightspace pages will have the navbar removed

**Comment by tarynm** [comment]
I had trouble trying to figure out a way to properly test the iframe content after the content is rendered. Not sure if you guys know of a way to do that. I added a test to test that the method to remove the navbar is called.

**Comment by jstefaniuk-d2l** [comment]
Could you set the content of the iframe with something like:
```
iframe.contentWindow.document.body.innerHTML = <div id="..."></div>
```
Then run the function and check that the `iframe.contentWindow.document.body.innerHTML` has changed in the desired way?

(there'll be a cleaner way to get the iframe's body, but I forget it)

**Comment by awikkerink** [comment]
That should work. I believe it is possible to manipulate the iframe with javascript if the src is not set

**Comment by tarynm** [comment]
iframes definitely make it extremely difficult to write tests for :(

**Comment by awikkerink** [comment]
LGTM :+1:

**Comment by jstefaniuk-d2l** [comment]
LGTM

**Comment by cgalvind2l** [comment]
LGTM

**Pull request merged**